.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,83 @@
+name: End 2 End Tests (Spaceflights Quickstart)
+
+on: # workflow_dispatch
+  push:
+     branches:
+       - develop
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Setup python
+      uses: actions/setup-python@v2.2.1
+      with:
+        python-version: 3.8
+
+    - name: Setup virtualenv
+      run: |
+        python -V
+        python -m pip install virtualenv
+        virtualenv venv
+        source venv/bin/activate
+   
+    - name: Initialize kedro spaceflights project
+      run: |
+        pip install . 'kedro<0.18'
+        kedro new --starter spaceflights --config tests/e2e/starter-config.yml --verbose
+
+    - name: Install project dependencies
+      run: |
+        cd ./spaceflights
+        echo "git+https://github.com/getindata/kedro-kubeflow.git@$GITHUB_SHA" >> src/requirements.txt
+        echo "kedro-docker" >> src/requirements.txt
+        sed -i '/kedro-telemetry/d' src/requirements.txt
+        cat src/requirements.txt
+        pip install -r src/requirements.txt
+
+    - name: Init and update configuration
+      run: |
+        cd ./spaceflights
+        kedro docker init
+        kedro kubeflow init http://localhost:9000
+        echo "!data/01_raw" >> .dockerignore
+        mv ../tests/e2e/catalog.yml conf/base/catalog.yml
+        rm -f conf/base/kubeflow.y*
+        mv ../tests/e2e/kubeflow.yml conf/base/kubeflow.yml
+
+    - name: Prepare docker env
+      uses: docker/setup-buildx-action@v1
+      id: buildx
+      with:
+        install: true
+
+    - name: Build pipeline docker image
+      run: |
+        cd ./spaceflights
+        docker build --build-arg BASE_IMAGE=python:3.8-buster --tag kedro-kubeflow-e2e:latest --load .
+
+    - name: Set up GCP Credentials
```

Comment:
nb: definitely not for now but maybe we can think about workload identity federation here? A little bit more setup but after that we don't have to deal with json keys (https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions)

Reply:
Pls make an issue out of this